### PR TITLE
Overwrite output

### PR DESCRIPTION
**Pull request by oligriffiths**
This PR adds an `--overwrite` CLI option to the build command.
It's very frustrating to have to remove the output directory before each build, so this fixes that issue.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -54,6 +55,7 @@ module.exports = function broccoliCLI(args) {
     .option('--output-path <path>', 'the path to target output folder')
     .option('--watch', 'turn on the watcher')
     .option('--watcher <watcher>', 'select sane watcher mode')
+    .option('--overwrite', 'overwrite the <target> directory')
```
Comment (chrmod) [inline lib/cli.js]: should probably refer to `<target>` by `output-path` name

**Comment by oligriffiths** [comment]
This will still prevent accidental deletions as it’s an optional flag. And I’m any case, the build should be deterministic, so you can run multiple times and the output is the same if the input doesn’t change.

**Comment by oligriffiths** [comment]
@chemod I like the idea of doing this automatically inside the project folder

**Comment by chrmod** [comment]
@oligriffiths one topic to consider is when broccoli is used from some script that inject output path from something like env variable. The output folder is protected from accidental deletion in case the passed value was unexpectedly change. Is such cases developer probably controls the script already, so you can run `rm -rf` on such variable.

Another alternative I see is to have .broccolirc file with a default params, so we can skip the guard.

**Comment by oligriffiths** [comment]
Also updated the PR so lint & tests pass

**Comment by oligriffiths** [comment]
@chrmod thoughts?

**Comment by oligriffiths** [comment]
bump

**Comment by oligriffiths** [comment]
I see in ember-cli that there is protection to ensure the directory to delete isn’t a parent of the project. Should probably add that.

**Comment by rwjblue** [comment]
Ya, good idea!

**Comment by rwjblue** [comment]
@oligriffiths - Wanna whip that up as a separate PR?

**Comment by oligriffiths** [comment]
Sure thinf

**Pull request merged**